src/Native/System.Native/pal_networking.h

```diff
@@ -90,6 +96,21 @@ struct IPPacketInformation
     int32_t Padding;        // Pad out to 8-byte alignment
 };
 
+struct IPv4MulticastOption
+{
+    uint32_t MulticastAddress; // Multicast address
+    uint32_t LocalAddress;     // Local address
+    int32_t InterfaceIndex;    // Interface index
+    int32_t Padding;           // Pad out to 8-byte alignment
+};
+
+struct IPv6MulticastOption
+{
+    IPAddress Address;      // Multicast address
+    int32_t InterfaceIndex; // Interface index
+    int32_t Padding;        // Pad out to 8-byte alignment
```

Comment:
No: @stephentoub and I spoke about this and decided that the explicit padding is necessary. We're going to have a single definition of the type in IL, so we must be sure that the native definition matches, even on platforms where the padding is unnecessary. We will need to rethink this in the remote chance that we encounter a platform that requires > 8 byte alignment.